documentation/reference/document-json-update-format.html

```diff
@@ -74,90 +89,205 @@ <h3 id="assign">assign</h3>
     }
 }
 </pre>
-<code>assign</code> can use fieldpath syntax when used on composite types when they are not part of attribute or index.
-On a <code>map</code> it can be used as a put.
-In the following example a tuple (key1, value1) is inserted into the map
-regardless if there is an element with key "key1" before:
+<h3 id="assign-struct-field">Struct field</h3>
+<h5>Replacing all fields in a struct</h5>
+<p>
+Entire struct fields may be replaced by assigning a JSON object of struct key/value pairs.
+</p>
+<pre>
+struct person {
+  field first_name type string {}
+  field last_name type string {}
+}
+field contact type person {
+  indexing: summary
+}
+</pre>
 <pre>
 {
-    "update": "id:music:music::example2",
+    "update": "id:mynamespace:workers::example",
     "fields": {
-        "mapfield{key1}": {
-            "assign": "value1"
+        "contact": {
+            "assign": {
+                "first_name": "Bob",
+                "last_name": "The Plumber"
+            }
         }
     }
 }
 </pre>
-<code>assign</code> can be used on structs to set one field
-<strong>Note:</strong> does not work for <a href="services-content.html#document">index</a> mode:
+<h5>Individual struct fields</h5>
+<p>
+Individual struct fields may be updated using <a href="#fieldpath">field path</a> syntax.
+</p>
 <pre>
 {
-    "update": "id:music:music::example3",
+    "update": "id:mynamespace:workers::example",
     "fields": {
-        "structfieldperson.firstname": { "assign": "bob" },
-        "structfieldperson.lastname": { "assign": "the plumber" }
+        "contact.first_name": { "assign": "Bob" },
+        "contact.last_name": { "assign": "The Plumber" }
     }
 }
 </pre>
-<code>assign</code> can be used on array to set one element. In this example we have an array of strings:
+<p>
+<strong>Note:</strong> this does not work for <em>attribute</em> struct fields.
+</p>
+
+<h3 id="assign-map-field">Map field</h3>
+<p>
+Individual map entries may be updated using <a href="document-field-path.html">field path</a> syntax.
+</p>
+<p>
+Assigning an element to a key in a map will insert the key/value mapping if does not already exist, or
```

Comment:
Typo? "...mapping if does not..." -> "...mapping if it does not"?

Reply:
Fixed